src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/SubmitVacancyCommandHandler.cs

```diff
@@ -5,29 +5,48 @@
 using MediatR;
 using System.Threading;
 using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Domain.Services;
 
 namespace Esfa.Recruit.Vacancies.Client.Application.CommandHandlers
 {
     public class SubmitVacancyCommandHandler : IRequestHandler<SubmitVacancyCommand>
     {
         private readonly IVacancyRepository _repository;
         private readonly IMessaging _messaging;
+        private readonly ITimeProvider _timeProvider;
 
-        public SubmitVacancyCommandHandler(IVacancyRepository repository, IMessaging messaging)
+        public SubmitVacancyCommandHandler(IVacancyRepository repository, IMessaging messaging, ITimeProvider timeProvider)
         {
             _repository = repository;
             _messaging = messaging;
+            _timeProvider = timeProvider;
         }
 
         public async Task Handle(SubmitVacancyCommand message, CancellationToken cancellationToken)
         {
-            await _repository.UpdateAsync(message.Vacancy);
+            var vacancy = await _repository.GetVacancyAsync(message.VacancyId);
```

Comment:
I guess the downside of this change is that we have to Get the vacancy more than once in the request.

Reply:
Yep. I spoke to Mark about this. Its a necessary performance hit as we can't be sure when the command is going to be executed (in the future this may be done out of process) so the potential for overwrites increases if we keep a handle to the full vacancy.  This way the editing time is shorter so the risk of overwrites diminishes.